CRM/Core/BAO/Address.php

```diff
@@ -379,14 +377,12 @@ public static function fixAddress(&$params) {
       }
     }
 
-    // check if geocode should be skipped (can be forced with an optional parameter through the api)
-    $skip_geocode = (isset($params['skip_geocode']) && $params['skip_geocode']) ? TRUE : FALSE;
-
-    // add latitude and longitude and format address if needed
-    if (!$skip_geocode && !empty($config->geocodeMethod) && ($config->geocodeMethod != 'CRM_Utils_Geocode_OpenStreetMaps') && empty($params['manual_geo_code'])) {
-      $class = $config->geocodeMethod;
-      $class::format($params);
+    // skip_geocode is an optional parameter through the api.
+    // manual_geo_code is on the contact edit form. They do the same thing....
+    if (empty($params['skip_geocode']) && empty($params['manual_geo_code'])) {
```

Comment:
@eileenmcnaughton, is it expanding scope too much to remove the form-layer logic from this class? Seems like the form should use the `skip_geocode` parameter rather than invent its own.

Reply:
I agree with you about the form - could just rename the field to 'skip_geocode'. I could possibly expand to it since we are both quite engaged I'm a bit less worried about increasing scope than I normally would be